Add Deco

schneidmaster opened the pull request:
- [X] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [X] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [X] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] Commit message includes cask’s name.
- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` left no offenses.
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.

Note: Cask uses `installer manual` rather than `pkg` because Deco.pkg runs a postinstall script that is [required to be run as the current user](https://github.com/decosoftware/deco-ide/blob/master/desktop/Scripts/postinstall#L5-L8), because the [postinstall script chowns some application files to $USER that would cause later runtime permissions errors if $USER was root](https://github.com/decosoftware/deco-ide/issues/6#issuecomment-221680139).

jonathanmarvens commented on Casks/deco.rb:
```diff
@@ -0,0 +1,18 @@
+cask 'deco' do
+  version :latest
+  sha256 :no_check
```
Why `:no_check`?

schneidmaster replied:
`brew cask audit --download {{cask_file}}` requires sha256 to be :no_check for casks that have an unversioned download link; otherwise, future updates to the pkg would break the homebrew-cask install without giving a chance to upgrade the cask metadata.

adidalal replied:
Needs to be, as URL is not versioned - if there is, this Cask can be modified accordingly

schneidmaster replied:
Yeah, there isn't a versioned URL anywhere I could find.

jonathanmarvens replied:
Ah, I figured that was the case. @schneidmaster: Did you check with the Deco creaters/maintainers to see if there are any versioned URLs?

schneidmaster replied:
Actually, I just discovered that appending a version like `https://s3-us-west-2.amazonaws.com/decopkgs/osx/Deco-0.6.0.pkg` will work. Updating PR.

jonathanmarvens replied:
@schneidmaster: Awesome!

adidalal commented on Casks/deco.rb:
```diff
@@ -0,0 +1,18 @@
+  version :latest
+  sha256 :no_check
+
+  url 'https://s3-us-west-2.amazonaws.com/decopkgs/osx/Deco.pkg'
```
Needs a `# s3-us-west-2.amazonaws.com/decopkgs was verified as official when first introduced to the cask` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

adidalal commented:
There should also be an appcast, as I see an autoupdater at https://github.com/decosoftware/deco-ide/blob/4474aac5cbf491663a0ad450568503cb167ed6ac/desktop/src/updateManager.js#L39

Should be able to use Charles/mitmproxy to get the URL if it can't be reconstructed from that .js

schneidmaster commented:
@adidalal I reconstructed the appcast url (`https://deco-nuts.herokuapp.com/update/osx/#{version}`), but the server doesn't seem to be up to date right now. The current version of Deco as downloaded from s3 is 0.6.0, but the appcast thinks the most recent version is 0.5.1, so I can't generate the checksum correctly. Advice?

adidalal commented:
That's fine, we don't actually pull the version from the URL but use it for helper scripts when batch updating casks. Add it in, with the checkpoint you currently get, it's not a problem.

schneidmaster commented:
Okay, so I added the appcast and comment about s3, and rebased back down to 1 commit. New problem is that the audit fails because "https://deco-nuts.herokuapp.com/update/osx/0.6.0" returns an empty string which sha256 doesn't want to hash. 

`cannot use the sha256 for an empty string: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`

adidalal commented:
Thank you for the contribution. It needed some fixes, so they were made in 2ddbfb4f74d7a0fea016178cd83085e200025dcf. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

adidalal commented:
appcast shasum doesn't matter, only the fact that it changes does (so any shasum is fine for now)

schneidmaster commented:
Ah cool. Thanks for the help.

schneidmaster commented:
@adidalal I noticed you changed the installer to `installer manual: 'Deco-#{version}.pkg'` -- you also need to change from single to double quotes (e.g. `installer manual: "Deco-#{version}.pkg"` so Ruby performs the string interpolation.

adidalal commented:
Updated, thanks.
